### PR TITLE
Add sql editing function

### PR DESCRIPTION
**Pull request by rivenwang**
The new api added includes the management addition, deletion, modification and query of sql file and the opening and closing of session, etc.

### CODE REVIEW

File: micro-api/controller/dataapps/sqleditor/sqleditor_api.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+// @Summary create session
+// @Schemes
+// @Description create session result
+// @Tags sqleditor
+// @Param cluster_id path string true "cluster ID"
+// @Param request_body body sqleditor.CreateSessionReq true "request_body"
+// @Accept json
+// @Produce json
+// @Success 200 {object} sqleditor.CreateSessionRes
+// @Router /{cluster_id}/session [post]
```
Comment (haiboumich) [inline micro-api/controller/dataapps/sqleditor/sqleditor_api.go:35]: maybe the @Router should start with '/dataapps/sqleditor/' ? Like '/dataapps/sqleditor/{cluster_id}/session'.
Same for the rest routers.

File: micro-cluster/dataapps/sqleditor/main_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package sqleditor
```
Comment (haiboumich) [inline micro-cluster/dataapps/sqleditor/main_test.go:1]: Please add copyright

File: micro-cluster/dataapps/sqleditor/manager_test.go [hunk 3]
```diff
@@ -0,0 +1,324 @@
+package sqleditor
```
Comment (haiboumich) [inline micro-cluster/dataapps/sqleditor/manager_test.go:1]: please add copyright

File: models/dataapps/sqleditor/dbagent/main_test.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package dbagent
```
Comment (haiboumich) [inline models/dataapps/sqleditor/dbagent/main_test.go:1]: please add copyright

File: models/dataapps/sqleditor/dbagent/readwriteimpl_test.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
+package dbagent
```
Comment (haiboumich) [inline models/dataapps/sqleditor/dbagent/readwriteimpl_test.go:1]: please add copyright

File: models/dataapps/sqleditor/dbagent/sessioncache.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
+package dbagent
```
Comment (haiboumich) [inline models/dataapps/sqleditor/dbagent/sessioncache.go:1]: please add copyright

File: models/dataapps/sqleditor/dbagent/sessioncache_test.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
+package dbagent
```
Comment (haiboumich) [inline models/dataapps/sqleditor/dbagent/sessioncache_test.go:1]: please add copyright

File: models/dataapps/sqleditor/sqlfile/main_test.go [hunk 8]
```diff
@@ -0,0 +1,63 @@
+/******************************************************************************
+ * Copyright (c)  2022 PingCAP                                                *
```
Comment (haiboumich) [inline models/dataapps/sqleditor/sqlfile/main_test.go:2]: 2023

File: models/dataapps/sqleditor/sqlfile/readwriteimpl_test.go [hunk 9]
```diff
@@ -0,0 +1,140 @@
+package sqleditorfile
```
Comment (haiboumich) [inline models/dataapps/sqleditor/sqlfile/readwriteimpl_test.go:1]: please add copyright

File: util/api/tidb/sql/clustermeta.go [hunk 10]
```diff
@@ -0,0 +1,251 @@
+/******************************************************************************
+ * Copyright (c)  2022 PingCAP                                                *
```
Comment (haiboumich) [inline util/api/tidb/sql/clustermeta.go:2]: 2023

**Review by haiboumich — CHANGES_REQUESTED** [review]

**Review by haiboumich — APPROVED** [review]
LGTM

**Comment by rivenwang** [comment]
merge

**Review by zhangpeijin-milo — APPROVED** [review]
LGTM

**Pull request merged**